Isinstance for non-builtin types fails to compile

### Acton Version

0.11.7.20221013.13.53.19

### Steps to Reproduce

Compile:
```
class Foo:
    pass

def f(x):
    if isinstance(x, Foo):
        pass
```

### Expected Behavior

Success

### Actual Behavior

```
Building project in /home/ddjoh/workspaces/act-test-1
  Compiling test.act for release
ERROR: internal compiler error: compilation of generated C code failed
NOTE: this is likely a bug in actonc, please report this at:
NOTE: https://github.com/actonlang/acton/issues/new?template=ice.md
NOTE: acton 0.11.7.20221013.13.53.19 compiled by ghc 8.10 on linux x86_64
NOTE: cc: cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0
cc stdout:

cc stderr:
In file included from /home/ddjoh/workspaces/acton/dist/builtin/__builtin__.h:3,
                 from /home/ddjoh/workspaces/acton/dist/builtin/builtin.h:16,
                 from act-test-1/out/types/test.h:2,
                 from act-test-1/out/types/test.c:1:
act-test-1/out/types/test.c: In function ‘test$$f’:
act-test-1/out/types/test.c:17:24: error: ‘Foo$methods’ undeclared (first use in this function); did you mean ‘$bool$methods’?
   17 |     if ($ISINSTANCE(x, Foo)->val) {
      |                        ^~~
/home/ddjoh/workspaces/acton/dist/builtin/common.h:35:65: note: in definition of macro ‘$ISINSTANCE’
   35 |                                while($c && $c != ($Super$class)&$T ## $methods) $c = $c->$superclass; \
      |                                                                 ^~
act-test-1/out/types/test.c:17:24: note: each undeclared identifier is reported only once for each function it appears in
   17 |     if ($ISINSTANCE(x, Foo)->val) {
      |                        ^~~
/home/ddjoh/workspaces/acton/dist/builtin/common.h:35:65: note: in definition of macro ‘$ISINSTANCE’
   35 |                                while($c && $c != ($Super$class)&$T ## $methods) $c = $c->$superclass; \
      |                                                                 ^~
```

--- a/compiler/Acton/CodeGen.hs
+++ b/compiler/Acton/CodeGen.hs
@@ -676,7 +676,7 @@ instance Gen Expr where
     gen env (Call _ e p _)          = genCall env [] e p
     gen env (Async _ e)             = gen env e
     gen env (TApp _ e ts)           = genInst env ts e
-    gen env (IsInstance _ e c)      = gen env primISINSTANCE <> parens (gen env e <> comma <+> gen env c)
+    gen env (IsInstance _ e c)      = gen env primISINSTANCE <> parens (gen env e <> comma <+> gen env (eQVar c))
     gen env (Dot _ e n)             = genDot env [] e n
     gen env (DotI _ e i)            = gen env e <> text "->" <> gen env componentsKW <> brackets (pretty i)
     gen env (RestI _ e i)           = gen env eNone <> semi <+> text "// CodeGen for tuple tail not implemented"

